include/nbl/video/IGPURayTracingPipeline.h

```diff
@@ -0,0 +1,107 @@
+#ifndef _NBL_I_GPU_RAY_TRACING_PIPELINE_H_INCLUDED_
+#define _NBL_I_GPU_RAY_TRACING_PIPELINE_H_INCLUDED_
+
+#include "nbl/asset/IPipeline.h"
+#include "nbl/asset/IRayTracingPipeline.h"
+
+#include "nbl/video/SPipelineCreationParams.h"
+
+
+namespace nbl::video
+{
+
+class IGPURayTracingPipeline : public IBackendObject, public asset::IRayTracingPipeline<const IGPUPipelineLayout, const IGPUShader>
+{
+        using pipeline_t = asset::IRayTracingPipeline<const IGPUPipelineLayout,const IGPUShader>;
+
+    public:
+
+        struct SShaderGroupHandle
+        {
+          private:
+            uint8_t data[video::SPhysicalDeviceLimits::ShaderGroupHandleSize];
+        };
+        static_assert(sizeof(SShaderGroupHandle) == video::SPhysicalDeviceLimits::ShaderGroupHandleSize);
+
+        struct SCreationParams final : pipeline_t::SCreationParams, SPipelineCreationParams<const IGPURayTracingPipeline>
+        {
+            #define base_flag(F) static_cast<uint64_t>(pipeline_t::SCreationParams::FLAGS::F)
+            enum class FLAGS : uint64_t
+            {
+                NONE = base_flag(NONE),
+                DISABLE_OPTIMIZATIONS = base_flag(DISABLE_OPTIMIZATIONS),
+                ALLOW_DERIVATIVES = base_flag(ALLOW_DERIVATIVES),
+                CAPTURE_STATISTICS = base_flag(CAPTURE_STATISTICS),
+                CAPTURE_INTERNAL_REPRESENTATIONS = base_flag(CAPTURE_INTERNAL_REPRESENTATIONS),
+                FAIL_ON_PIPELINE_COMPILE_REQUIRED = base_flag(FAIL_ON_PIPELINE_COMPILE_REQUIRED),
+                EARLY_RETURN_ON_FAILURE = base_flag(EARLY_RETURN_ON_FAILURE),
+                LINK_TIME_OPTIMIZATION = base_flag(LINK_TIME_OPTIMIZATION),
+                RETAIN_LINK_TIME_OPTIMIZATION_INFO_BIT_EXT = base_flag(RETAIN_LINK_TIME_OPTIMIZATION_INFO),
+                SKIP_TRIANGLES = base_flag(SKIP_TRIANGLES),
+                SKIP_AABBS = base_flag(SKIP_AABBS),
+                NO_NULL_ANY_HIT_SHADERS = 1<<14,
+                NO_NULL_CLOSEST_HIT_SHADERS = 1<<15,
+                NO_NULL_MISS_SHADERS = 1<<16,
+                NO_NULL_INTERSECTION_SHADERS = 1<<17,
+                ALLOW_MOTION = 1<<20,
+            };
+            #undef base_flag
+
+            inline SSpecializationValidationResult valid() const
+            {
+                if (!layout)
+                  return {};
+
+                SSpecializationValidationResult retval = {
+                    .count=0,
+                    .dataSize=0,
+                };
```

Comment:
need to do some validation on the flags such as https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#VUID-VkRayTracingPipelineCreateInfoKHR-flags-06546